tests/monitoring_test.go

```diff
@@ -186,6 +229,102 @@ var _ = Describe("[Destructive] Monitoring Tests", func() {
 			waitForPrometheusAlert(f, "CDIMultipleDefaultVirtStorageClasses")
 		})
 
+		It("[test_id:XXXX]CDINoDefaultStorageClass fired when no default storage class exists, and a DataVolume is waiting for one", func() {
+			By("Ensure initial metric values")
+			defaultSCs := getMetricValue(f, "kubevirt_cdi_default_storageclasses")
+			Expect(defaultSCs).To(Equal(1))
+			defaultVSCs := getMetricValue(f, "kubevirt_cdi_default_virt_storageclasses")
+			Expect(defaultVSCs).To(BeNumerically("<=", 1))
+			waitingDVs := getMetricValue(f, "kubevirt_cdi_datavolumes_pending_for_default_storageclass")
+			Expect(waitingDVs).To(BeZero())
+
+			updateDefaultStorageClasses("false")
+
+			dv := utils.NewDataVolumeWithHTTPImportAndStorageSpec("test-dv", "1Gi", fmt.Sprintf(utils.TinyCoreQcow2URL, f.CdiInstallNs))
+			_, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dv)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Ensure metric values after removing default storage class annotation")
+			Eventually(func() int {
+				return getMetricValue(f, "kubevirt_cdi_default_storageclasses")
+			}, metricPollingTimeout, metricPollingInterval).Should(BeZero())
+			Eventually(func() int {
+				return getMetricValue(f, "kubevirt_cdi_default_virt_storageclasses")
+			}, metricPollingTimeout, metricPollingInterval).Should(BeZero())
+			Eventually(func() int {
+				return getMetricValue(f, "kubevirt_cdi_datavolumes_pending_for_default_storageclass")
+			}, metricPollingTimeout, metricPollingInterval).Should(Equal(1))
+
+			By("Check CDINoDefaultStorageClass alert is triggered")
+			waitForPrometheusAlert(f, "CDINoDefaultStorageClass")
+
+			updateDefaultStorageClasses("true")
+
+			By("Ensure metric values after restoring default storage class annotation")
+			Eventually(func() int {
+				return getMetricValue(f, "kubevirt_cdi_default_storageclasses")
+			}, metricPollingTimeout, metricPollingInterval).Should(Equal(defaultSCs))
+			Eventually(func() int {
+				return getMetricValue(f, "kubevirt_cdi_default_virt_storageclasses")
+			}, metricPollingTimeout, metricPollingInterval).Should(Equal(defaultVSCs))
+			Eventually(func() int {
+				return getMetricValue(f, "kubevirt_cdi_datavolumes_pending_for_default_storageclass")
+			}, metricPollingTimeout, metricPollingInterval).Should(BeZero())
+		})
+
+		It("[test_id:XXXX]CDIDefaultStorageClassDegraded fired when default storage class has no smart clone or ReadWriteMany", func() {
+			if f.SnapshotSCName != defaultStorageClass.Name && f.CsiCloneSCName != defaultStorageClass.Name {
+				Skip("Default storage class does not support snapshot or CSI clone")
+			}
+			if !f.IsSnapshotStorageClassAvailable() && !f.IsCSIVolumeCloneStorageClassAvailable() {
```

Comment:
I think this test will always get skipped since the destructive lane only has HPP (no snapshot support)
Maybe since we don't actually need snapshot functionality, just the vsc object, we can set it up ad hoc just for this test?

Reply:
Right, I'll try your nice idea :)